src/test/java/com/exadel/discount/controller/DiscountTests.java

```diff
@@ -0,0 +1,197 @@
+    }
+
+    @Test
+    @Order(4)
+    void findAllDiscountsWithFiltersIT() throws Exception {
+        given()
+                .webAppContextSetup(wac)
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON)
+                .param("categoryId", "93577f24-f68f-403e-aa04-0a60c3a445d6")
+                .param("vendorId", "93577f24-f68f-403e-aa04-0a60c3a445d2")
+                .param("vendorLocationsIds", "93577f24-f68f-403e-aa04-0a60c3a445d3",
+                        "93577f24-f68f-403e-aa04-0a60c3a445d4")
+                .param("tagIds", "93577f24-f68f-403e-aa04-0a60c3a445d65")
+                .param("LocalDateTime endDateTimeFrom", "2021-07-19T23:50:00")
+                .param("LocalDateTime endDateTimeTo", "2021-07-20T00:10:00")
+                .when()
+                .get("/discounts")
+                .then()
+                .assertThat(MockMvcResultMatchers.status().isOk())
+                .assertThat(MockMvcResultMatchers.status().is2xxSuccessful())
+                .assertThat(jsonPath("$[0].name").value("Toy for your dog"))
+                .assertThat(jsonPath("$[0].description").value("Dog stuff"))
+                .assertThat(jsonPath("$[0].value").value("5"))
+                .assertThat(jsonPath("$[0].discountType").value("PERCENT"))
+                .assertThat(jsonPath("$[0].promo").value("1abcde1"))
+                .assertThat(jsonPath("$[0].endTime").value("2021-07-20T00:00:00"))
+                .assertThat(jsonPath("$[0].archived").value("false"))
+                .assertThat(jsonPath("$[0].vendor.name").value("Dog stuff"))
+                .assertThat(jsonPath("$[0].category.name").value("Dogs"));
+    }
+
+    @Test
+    @Order(5)
+    public void getDiscountSearchingIT() throws Exception {
+        given().webAppContextSetup(wac)
+                .contentType(ContentType.JSON)
+                .accept(ContentType.JSON)
+                .param("searchText", "Toy for")
+                .when()
+                .get("/discounts/search")
+                .then()
+                .assertThat(MockMvcResultMatchers.status().isOk())
+                .assertThat(MockMvcResultMatchers.status().is2xxSuccessful())
+                .assertThat(jsonPath("$[0].name").value("Toy for your dog"))
+                .assertThat(jsonPath("$[0].description").value("Dog stuff"))
+                .assertThat(jsonPath("$[0].value").value("5"))
+                .assertThat(jsonPath("$[0].discountType").value("PERCENT"))
+                .assertThat(jsonPath("$[0].promo").value("1abcde1"))
+                .assertThat(jsonPath("$[0].endTime").value("2021-07-20T00:00:00"))
+                .assertThat(jsonPath("$[0].archived").value("false"))
+                .assertThat(jsonPath("$[0].vendor.name").value("Dog stuff"))
+                .assertThat(jsonPath("$[0].category.name").value("Dogs"));
+    }
+
+    @Test
+    @Order(6)
+    public void deleteDiscountAndCheckIT() throws Exception {
```

Comment:
this test actually consists of 3 tests(test delete, test if archived and test restore.),  Please name it properly or split it into 3 tests